platforms/software/ivy/src/main/java/org/gradle/api/publish/ivy/tasks/PublishToIvyRepository.java

```diff
@@ -127,29 +116,17 @@ private static IvyPublicationInternal toPublicationInternal(IvyPublication publi
     /**
      * The repository to publish to.
      *
-     * @return The repository to publish to
+     * Only instances of DefaultIvyArtifactRepository are supported
      */
     @Internal
-    @ToBeReplacedByLazyProperty
-    public IvyArtifactRepository getRepository() {
+    @ReplacesEagerProperty
+    public Property<IvyArtifactRepository> getRepository() {
```

Comment:
❓ Also this, do we want to have this as Property? It also looks like a nested property
cc @jvandort